### PR TITLE
settings: add network settings extension

### LINKED ISSUE #3652: OOTB: Port `network` settings model to settings extensions
Create a settings extension for `network`.

### Definition of done
A new `network` crate is created under `sources/settings-extensions`, and all variants use the model defined in that crate in place of the existing settings. Additionally, a basic `settings-network` RPM package is created that installs the extension binary.

### PR DESCRIPTION
**Pull request by sam-berning**
**Issue number:**

Closes #3652 

**Description of changes:**

Creates a network settings extension

**Testing done:**

Unit testing, built `aws-dev` variant and packaged `settings-network` in it. Tested on an ec2 instance.

network settings worked as expected, and the setting extension generation returned as expected:

```
bash-5.1# /usr/libexec/settings/network proto1 generate --setting-version v1
{
  "Complete": {
    "hostname": "ip-{instance public ip}.us-west-2.compute.internal"
  }
}
```

(EDIT 1-18)

Also built and tested a `vmware-dev` image, and the extension worked as expected:

```
bash-5.1# /usr/libexec/settings/network proto1 generate --setting-version v1
{
  "Complete": {
    "hostname": "{vm ip address}"
  }
}
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/settings-extensions/network/src/lib.rs
Comment (cbgbt) [inline sources/settings-extensions/network/src/lib.rs]: This should be expressed as an RPM dependency.

File: packages/settings-network/Cargo.toml
Comment (cbgbt) [inline packages/settings-network/Cargo.toml:21]: As mentioned elsewhere, we'll want to add the netdog RPM as a dependency here.

File: packages/settings-network/settings-network.spec
Comment (cbgbt) [inline packages/settings-network/settings-network.spec:13]: Likewise, we want to add a `Requires: %{_cross_os}netdog`

This symbol would only be available once we merge in https://github.com/bottlerocket-os/bottlerocket/pull/3700

**Review by cbgbt — APPROVED** [review]
This looks right to me, other than the need to express a dependency on netdog in the RPM. Since this requires https://github.com/bottlerocket-os/bottlerocket/pull/3700, we should either open an issue or merge this after that PR has landed.

File: packages/settings-network/settings-network.spec
Comment (webern) [reply packages/settings-network/settings-network.spec:13]: > Likewise, we want to add a Requires: %{_cross_os}netdog
>
> This symbol would only be available once we merge in https://github.com/bottlerocket-os/bottlerocket/pull/3700

Isn't it the other way around? Doesn't `netdog` depend on its settings extension?

Edit: oh I see, this extension calls out to `netdog` makes sense. This is a setting that relies on netdog to find its value.

File: sources/models/Cargo.toml
Comment (webern) [inline sources/models/Cargo.toml]:
```suggestion
settings-extension-container-registry = { path = "../settings-extensions/container-registry", version = "0.1" }
settings-extension-network = { path = "../settings-extensions/network", version = "0.1" }
settings-extension-ntp = { path = "../settings-extensions/ntp", version = "0.1" }
settings-extension-updates = { path = "../settings-extensions/updates", version = "0.1" }
```

Comment (webern) [reply sources/models/Cargo.toml]: Maybe keep these sorted?

File: sources/settings-extensions/network/src/generate.rs
Comment (webern) [inline sources/settings-extensions/network/src/generate.rs]: Can the braces be eliminated in these zero params Snafu struct constructions? (can't remember)

```suggestion
        Ok(ValidLinuxHostname::try_from(hostname).context(InvalidHostnameSnafu)?)
```

File: sources/settings-extensions/network/src/lib.rs
Comment (webern) [inline sources/settings-extensions/network/src/lib.rs:52]:
```suggestion
        // This settings field types handle validation during deserialization.
        Ok(())
```

File: sources/settings-extensions/network/src/lib.rs
Comment (webern) [inline sources/settings-extensions/network/src/lib.rs:76]: Add a test where the JSON fails to parse due to an invalid hostname or some other invalid field. It helps document the way the type enforces correctness of its input.

**Review by webern — APPROVED** [review]

File: sources/settings-extensions/network/src/generate.rs
Comment (sam-berning) [reply sources/settings-extensions/network/src/generate.rs]: looks like they can! updating

**Pull request closed without merging**
